src/missile/missile.cpp

```diff
@@ -1297,6 +1297,19 @@ void CleanMissiles()
 	LocalMissiles.clear();
 }
 
+template <typename Range>
+bool is_PercentSortedDecreasing(Range &range)
+{
+	return ranges::is_sorted(
+		range,
+		[](const auto &lhs, const auto &rhs) { return lhs.Percent > rhs.Percent; });
+}
+
+bool is_BurningBuildingFramesSorted()
```

Comment:
I would change Sorted by "OK"/"Valid" and check also that first percent is 0, and last <= 100.

Reply:
done